src/Carnac/CarnacTrayIcon.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Drawing;
+using System.Reflection;
+using System.Windows.Forms;
+using Application = System.Windows.Application;
+
+namespace Carnac
+{
+    public class CarnacTrayIcon : IDisposable
+    {
+        readonly NotifyIcon trayIcon;
+
+        public CarnacTrayIcon()
+        {
+            var exitMenuItem = new MenuItem
+            {
+                Text = Properties.Resources.ShellView_Exit
+            };
+
+            var iconStream = Assembly.GetExecutingAssembly().GetManifestResourceStream("Carnac.icon.embedded.ico");
+
+            trayIcon = new NotifyIcon
+            {
+                Icon = new Icon(iconStream),
+                ContextMenu = new ContextMenu(new[] { exitMenuItem })
+            };
+
+            exitMenuItem.Click += (sender, args) =>
+            {
+                trayIcon.Visible = false;
+                Application.Current.Shutdown();
+            };
+            trayIcon.DoubleClick += NotifyIconClick;
```

Comment:
I thought I was going crazy testing this, but can we put this back to a single-click?

```
ni.MouseClick += NotifyIconClick;
```